Assets/LoggingManager/LoggingManager.cs

```diff
@@ -126,220 +95,147 @@ public void SaveLog(string collectionLabel)
-        string headerLine = "";
-        if (collections[label].saveHeaders)
+        if (logsList.ContainsKey(collectionLabel))
         {
-            headerLine = GenerateHeaders(collections[label]);
+            logsList.Remove(collectionLabel);
         }
-        object temp;
-        string filename = collections[label].label;
-        string filePath = savePath + "/" + filePrefix + filestamp + filename + fileExtension;
-        using (var file = new StreamWriter(filePath, true))
+        else
         {
-            if (collections[label].saveHeaders)
-            {
-                file.WriteLine(headerLine);
-                collections[label].saveHeaders = false;
-            }
-            for (int i = 0; i <= collections[label].count; i++)
-            {
-                string line = "";
-                foreach (KeyValuePair<string, Dictionary<int, object>> log in collections[label].log)
-                {
-                    if (line != "")
-                    {
-                        line += fieldSeperator;
-                    }
-
-                    if (log.Value.TryGetValue(i, out temp))
-                    {
-                        line += ConvertToString(temp);
-                    }
-                    else
-                    {
-                        line += "NULL";
-                    }
-                }
-                file.WriteLine(line);
-            }
+            Debug.LogError("Collection " + collectionLabel + " does not exist.");
         }
-        Debug.Log(label + " logs with " + collections[label].count + 1 + " rows saved to " + savePath);
     }
 
 
-    // Generates the headers in a CSV format and saves them to the CSV file
-    private string GenerateHeaders(LogCollection collection)
+    public void DeleteAllLogs()
     {
-        string headers = "";
-        foreach (string key in collection.log.Keys)
+        foreach (var keyValuePair in logsList)
         {
-            if (headers != "")
-            {
-                headers += fieldSeperator;
-            }
-            headers += key;
+            logsList.Remove(keyValuePair.Key);
         }
-        return headers;
     }
 
-    private void SaveToSQL(string label)
+    // Formats the logs to a CSV row format and saves them. Calls the CSV headers generation beforehand.
+    // If a parameter doesn't have a value for a given row, uses the given value given previously (see 
+    // UpdateHeadersAndDefaults).
+    private void SaveToCSV(string label)
     {
-        if (!enableMySQLSave) { return; }
-
-        if (!collections.ContainsKey(label))
+        if (!enableCSVSave) return;
+        if (logsList.TryGetValue(label, out LogStore logStore))
         {
-            Debug.LogError("Could not find collection " + label + ". Aborting.");
-            return;
+            WriteToCSV writeToCsv = new WriteToCSV(logStore, savePath, filePrefix, fileExtension);
+            writeToCsv.WriteAll();
         }
+    }
```

Comment:
nice.

Maybe throw an error or a `Debug.LogWarning()` if the user is trying to save a logCollection which doesnt exist?